File: Week 7/id_445/231.2-的幂.cpp
/*
 * @lc app=leetcode.cn id=231 lang=cpp
 *
 * [231] 2的幂
 */

// @lc code=start
class Solution {
public:
    bool isPowerOfTwo(int n) {
        // if(n<=0) return false;
        // if(n==-2147483648)return true;
        return n>0&&(n&(n-1))==0;
    }
};
// @lc code=end

File: Week 7/id_520/231.2-的幂.java
/*
 * @lc app=leetcode.cn id=231 lang=java
 *
 * [231] 2的幂
 */

// @lc code=start
class Solution {
    public boolean isPowerOfTwo(int n) {
        // 如果n是2的幂次，那么只有一位是1，去掉它，整个就是0了
        // 注意前提，n 要不为0
        return n > 0 && (n&(n-1))==0;
    }
}
// @lc code=end

